src/coins.cpp

```diff
@@ -214,12 +214,29 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
 }
 
 bool CCoinsViewCache::Flush() {
-    bool fOk = base->BatchWrite(cacheCoins, hashBlock);
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase*/ true);
     cacheCoins.clear();
     cachedCoinsUsage = 0;
     return fOk;
 }
 
+bool CCoinsViewCache::Sync()
+{
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase*/ false);
+    // Instead of clearing the cache, just clear the FRESH/DIRTY
+    // flags, and erase any spent coins
+    for (auto it = cacheCoins.begin(); it != cacheCoins.end(); ) {
+        it->second.flags = 0;
```

Comment:
nit: it seems a bit odd to clear the flags immediately before deleting in the case where the coin is spent. If you retouch this PR, the flag clearing can be moved into the else branch below.

You might also want to update the comment to reflect that change: "Erase any spent coins and clear the FRESH/DIRTY flags for unspent coins".

Reply:
Agreed. In addition, let's not assume there won't be new flags in the future, so maybe clear the `FRESH/DIRTY` flags explicitly with an `=&`?

Reply:
fixed